src/validation.ts

```diff
@@ -703,49 +770,65 @@ export const snakeCaseTransformation = (key: string): string =>
     .replace(lowerThenUpper, '$1_$2')
     .toLowerCase()
 
+function withLogicalType<T>(
```

Comment:
do we really need logicalType ? It seems there are two concepts: the input tag, and the output (final) tag, after all the mappings and transformations took place. For now, we only really need the input one so we could name it like that?

`meta.inputTag`

Reply:
[Logical type](https://avro.apache.org/docs/current/spec.html#Logical+Types) is how the input is validated (string as date, string as url, ...), for simple transformation.

Not mandatory for the use case, only think it's useful, and not hard to expose as metadata.

Reply:
Yeah I understand now. tag is the basic type (like string) and logicalType has extra info. Let's keep it.